torchgeo/datasets/geo.py

```diff
@@ -715,18 +715,19 @@ def __init__(
             match = re.match(filename_regex, os.path.basename(filepath))
             if match is not None:
                 try:
-                    with fiona.open(filepath, layer=layer) as src:
-                        if crs is None:
-                            crs = CRS.from_wkt(src.crs_wkt)
-
-                        minx, miny, maxx, maxy = src.bounds
-                        (minx, maxx), (miny, maxy) = fiona.transform.transform(
-                            src.crs, crs.to_wkt(), [minx, maxx], [miny, maxy]
+                    src = gpd.read_file(filepath, layer=layer)
+                    if src.crs is None:
+                        crs = CRS.from_wkt(src.crs_wkt)
+                        src.to_crs(crs, inplace=True)
+                        geometries = src.bounds
+                        geometries.apply(
+                            lambda row: shapely.geometry.box(
+                                row['minx'], row['miny'], row['maxx'], row['maxy']
+                            ),
+                            axis=1,
                         )
-                        geometry = shapely.box(minx, miny, maxx, maxy)
-                        geometries.append(geometry)
-                except fiona.errors.FionaValueError:
-                    # Skip files that fiona is unable to read
+                except pyogrio.errors.DataSourceError:
```

Comment:
@adamjstewart unfortunately geopandas doesn't have it's own error that it throws when it can't read a file, it defaults to letting `pyogrio` throw an error. Basically, would have to add `pyogrio` as a dependency just for this error...

Reply:
`DataSourceError` is a subclass of `RuntimeError`, just catch that instead.